dialogue-target/src/main/java/com/palantir/dialogue/Request.java

```diff
@@ -82,6 +83,10 @@ private Request(Builder builder) {
         return body;
     }
 
+    public Optional<CallingThreadExecutor> getCallingThreadExecutor() {
```

Comment:
A bit sucky this is here, but it's good enough. The problem is that queuing is cooperative, so when we finish a request, we use that thread to schedule the next request in the queue to run. This means we need to be able to then jump back to the calling thread for each request.